### PR TITLE
port file: add host name

### LINKED ISSUE #1050: cylc discovery commands should look for suite in configured suite hosts
`cylc gsummary` already does this. `cylc gui`, `cylc ping`, `cylc scan`, etc should do this as well.

### LINKED ISSUE #1285: port files should contain host name
On a facility with a shared filesystem (such as at NIWA), the port file (visible from all hosts) prevents a user from accidentally starting a suite that is already running on another host, but (a) it doesn't tell the user which host the suite is already running on, and (b) if a suite suffers a hard kill, the user may have to use "cylc scan" and/or "ps" on multiple hosts before manually deleting a left-over port file.

So I think it would be useful to have suite port files contain host name as well as port number.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Also:
* on 1st submit to a remote job host, install `passphrase` file (with `cylc-suite-env`) to its run directory.
* move logic from `cylc.port_file` to `cylc.scheduler` (write, unlink) and `cylc.network.pyro_base` (read).
* print suite host on suite start.
* print `host:port` to log on suite start and to GUI title.

Close #1050. Close #1285.

(Ideally, we should rename *port file* to *server location file* or something like that, and write `host:port\n` instead of `port\nhost\n` to the file, but the current approach is probably the most backward compatible.)

### CODE REVIEW

**Comment by matthewrmshin** [comment]
(All command line tools appear to work fine. I'll test the GUIs tomorrow.)

**Comment by matthewrmshin** [comment]
GUI looks good. New test added.

**Comment by matthewrmshin** [comment]
@hjoliver @arjclark This is now ready for review and test.

**Comment by matthewrmshin** [comment]
(Branch squashed. Improve GUI's call out to cylc commands - only add `--host=HOST` if necessary.)

**Comment by hjoliver** [comment]
(looks good, but I still need to test it...)

**Comment by matthewrmshin** [comment]
(Branch re-based.)

File: lib/cylc/CylcOptionParsers.py [hunk 1]
```diff
@@ -134,8 +134,7 @@ def add_std_options(self):
                 "--host",
                 help="Other host name. This results in "
                 "command reinvocation on the remote account.",
-                metavar="HOST", action="store", default=get_hostname(),
-                dest="host")
+                metavar="HOST", action="store", dest="host")
```
Comment (hjoliver) [inline lib/cylc/CylcOptionParsers.py]: The get_hostname import can be removed now.

Comment (matthewrmshin) [inline lib/cylc/CylcOptionParsers.py]: Done.

File: lib/cylc/gui/app_gcylc.py [hunk 2]
```diff
@@ -609,8 +609,9 @@ def reset(self, suite):
         self.cfg.reset(suite)
 
         win_title = suite
-        if self.cfg.host != socket.getfqdn():
-            win_title += " - %s" % self.cfg.host
+        if (self.cfg.host is not None and self.cfg.port is not None and
+                is_remote_host(self.cfg.host)):
+            win_title += " - %s:%d" % (self.cfg.host, int(self.cfg.port))
```
Comment (hjoliver) [inline lib/cylc/gui/app_gcylc.py]: The `socket` import can be removed.

Comment (matthewrmshin) [inline lib/cylc/gui/app_gcylc.py]: Done.

File: lib/cylc/scheduler.py [hunk 3]
```diff
@@ -714,7 +740,7 @@ def configure_suite(self, reconfigure=False):
             else:
                 # Remove database created by previous runs
                 if os.path.isdir(pri_db_path):
-                    shutil.rmtree(pri_db_path)
+                    rmtree(pri_db_path)
```
Comment (hjoliver) [inline lib/cylc/scheduler.py]: Good spotting ... but why is/was this test necessary (how can `pri_db_path` be a dir?)

**Comment by hjoliver** [comment]
@matthewrmshin - I still have a bit of testing to do on this, sorry (xmas rush...).  Note I'm on leave Wednesday, but working Thursday - will finish this then.  In the meantime @arjclark can do his review if that suits.

File: lib/cylc/scheduler.py (hunk 3 above)
Comment (matthewrmshin) [inline lib/cylc/scheduler.py]: It think it was probably an overly cautious test. No, it should never happen in real life (but one can never be too careful when there are users!)

Comment (hjoliver) [inline lib/cylc/scheduler.py]: Fair point!

**Comment by matthewrmshin** [comment]
@hjoliver Thanks for looking at this.

**Comment by arjclark** [comment]
Getting problems with one of the tests vs. master:
```
[INFO] cd "/path/to/home/cylc.git"
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that the RSA host key has just been changed.
The fingerprint for the RSA key sent by the remote host is
0a:14:59:7f:68:e3:e1:5c:28:c0:b5:4c:b5:35:62:20.
Please contact your system administrator.
Add correct host key in /path/to/home/.ssh/known_hosts to get rid of this message.
Offending key in /path/to/home/.ssh/known_hosts:126
RSA host key for localhost has changed and you have requested strict checking.
Host key verification failed.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 12/12 subtests 

Test Summary Report
-------------------
./tests/cylc-scan/01-hosts.t (Wstat: 256 Tests: 0 Failed: 0)
  Non-zero exit status: 1
  Parse errors: Bad plan.  You planned 12 tests but ran 0.
Files=1, Tests=0,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.20 cusr  0.07 csys =  0.30 CPU)
Result: FAIL
```

**Comment by arjclark** [comment]
Now prints out the same as above but multiple times followed by:

```
ok 1 - 01-hosts
not ok 2 - 01-hosts.stdout-grep-ok

    stdout and stderr stored in: /var/tmp/cylc-tests-me/cylc-scan/01-hosts-20151222T140143Z
Dubious, test returned 1 (wstat 256, 0x100)
Failed 11/12 subtests 

Test Summary Report
-------------------
./tests/cylc-scan/01-hosts.t (Wstat: 256 Tests: 2 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
  Parse errors: Bad plan.  You planned 12 tests but ran 2.
Files=1, Tests=2, 143 wallclock secs ( 0.03 usr  0.00 sys + 17.80 cusr  9.32 csys = 27.15 CPU)
Result: FAIL
```
From the logs:
```
cat /var/tmp/cylc-tests-me/cylc-scan/01-hosts-20151222T140143Z/01-hosts.stdout-grep-ok.stderr 
Can't find ^83b6d66c-816c-429f-9b92-ab9c4e54062c-localhost in 01-hosts.stdout
```

**Comment by arjclark** [comment]
Now working with the ssh/scp changes and increased timeout.

**Comment by arjclark** [comment]
Review 2: Looks OK to me. Test battery now passing in my environment.

**Comment by hjoliver** [comment]
No problems found, tests all pass here.

**Pull request merged**